internal/wal/wal_aof.go

```diff
@@ -305,7 +308,82 @@ func (wal *AOF) deleteSegmentPeriodically() {
 	}
 }
 
-func (wal *AOF) ForEachCommand(f func(c cmd.DiceDBCmd) error) error {
-	// TODO: implement this method
+func (wal *AOF) getSegmentFiles() ([]string, error) {
+	// Get all segment files matching the pattern
+	files, err := filepath.Glob(filepath.Join(wal.logDir, segmentPrefix+"*"+segmentSuffix))
+	if err != nil {
+		return nil, err
+	}
+
+	// Sort files by numeric suffix
+	sort.Slice(files, func(i, j int) bool {
+		parseSuffix := func(name string) int64 {
+			num, _ := strconv.ParseInt(
+				strings.TrimPrefix(strings.TrimSuffix(filepath.Base(name), segmentSuffix), segmentPrefix), 10, 64)
+			return num
+		}
+		return parseSuffix(files[i]) < parseSuffix(files[j])
+	})
+
+	return files, nil
+}
+
+func (wal *AOF) ReplayWAL(callback func(*WALEntry) error) error {
+	// Get list of segment files sorted by timestamp
+	segments, err := wal.getSegmentFiles()
+	if err != nil {
+		return fmt.Errorf("error getting segment files: %w", err)
+	}
+
+	// Process each segment file in order
+	for _, segment := range segments {
+		file, err := os.Open(segment)
+		if err != nil {
+			return fmt.Errorf("error opening segment file %s: %w", segment, err)
+		}
+
+		reader := bufio.NewReader(file)
+		for {
+			// Read entry size
+			var entrySize int32
+			if err := binary.Read(reader, binary.LittleEndian, &entrySize); err != nil {
+				if err == io.EOF {
+					break
+				}
+				file.Close()
+				return fmt.Errorf("error reading entry size: %w", err)
+			}
+
+			// Read entry data
+			entryData := make([]byte, entrySize)
+			if _, err := io.ReadFull(reader, entryData); err != nil {
+				file.Close()
+				return fmt.Errorf("error reading entry data: %w", err)
+			}
+
+			// Unmarshal entry
+			var entry WALEntry
+			MustUnmarshal(entryData, &entry)
+
+			// Call provided replay function with parsed command
+			if err := wal.ForEachCommand(&entry, callback); err != nil {
+				file.Close()
+				return fmt.Errorf("error replaying command: %w", err)
+			}
+		}
+		file.Close()
+	}
+
 	return nil
 }
+
+func (wal *AOF) ForEachCommand(entry *WALEntry, callback func(*WALEntry) error) error {
+	// Validate CRC
+	expectedCRC := crc32.ChecksumIEEE(append(entry.Data, byte(entry.LogSequenceNumber)))
+	if entry.Crc32 != expectedCRC {
+		return fmt.Errorf("CRC mismatch for log sequence %d: expected %d, got %d",
```

Comment:
refer to [LOGGING](https://github.com/DiceDB/dice/blob/master/CONTRIBUTING/logging.md) practices and apply them in this entire patch.